### ISSUE TITLE
error while building re

### ISSUE THREAD

**Issue by Drup** [OlivierNicole/ocaml#6]
```
File "lib/re.ml", line 1:
Error: Error while linking /tmp/camlstatic1247ad.cmm:
Reference to undefined global `Re_automata'
```

With a fresh macro switch (5 minutes ago, 27b6d8e).

**Comment by OlivierNicole** [comment]
I assume you are trying to compile an Opam package or some other project? Unfortunately the linker, as it works currently, makes a very bad approximation of the dependency tree, i.e. each runtime dependency of a module, becomes a static dependency, and static dependencies have to be compiled before their parent node in the tree.

It really sucks and I've started fixing this today, but it should be available only later this week, tomorrow at best.

**Comment by OlivierNicole** [comment]
I just read the title of the issue. So yeah you might want to read the end of the README about what makes a lot of packages broken currently. Once static dependencies are fixed, hopefully most packages will work again.

**Comment by Drup** [comment]
Yes, I'm talking about [re](https://github.com/ocaml/ocaml-re). To reproduce, just try to compile it.

**Comment by OlivierNicole** [comment]
The dependency issue appears fixed at revision c9bc6febde96c6a5c1e854d3c610d9a92f46f192 of macros_unstable. re compiles again, and so should most packages. I still need to fix a few tests before merging it into the main branch.

**Comment by OlivierNicole** [comment]
re should compile again at latest revision of the `macros` branch.
